OsmAnd/src/net/osmand/plus/auto/FavoritesScreen.java

```diff
@@ -60,6 +62,14 @@ public FavoritesScreen(
 		this.settingsAction = settingsAction;
 		this.surfaceRenderer = surfaceRenderer;
 		selectedGroup = group;
+		getLifecycle().addObserver(new DefaultLifecycleObserver() {
+			@Override
+			public void onDestroy(@NonNull LifecycleOwner owner) {
+				DefaultLifecycleObserver.super.onDestroy(owner);
+				getApp().getOsmandMap().getMapLayers().getFavouritesLayer().setAndroidAutoFavouritePoints(null);
+				getApp().getOsmandMap().refreshMap();
```

Comment:
Probably we should move this two rows in separate methods in each screen, also add getOsmandMap in base screen class